src/dashboard/src/components/helpers.py

```diff
@@ -334,3 +337,11 @@ def default_processing_config_path():
         get_server_config_value('sharedDirectory'),
         'sharedMicroServiceTasksConfigs/processingMCPConfigs/defaultProcessingMCP.xml'
     )
+
+def clean_date(date):
+    if date:
+        try:
+            return dateutil.parser.parse(date).strftime('%Y-%m-%d')
+        except ValueError:
+            raise forms.ValidationError('Unable to interpret as a date: %(date)s',
+                                        params={'date': date})
```

Comment:
`return date` in all cases?  Django clean functions say to always return data, and it may be something Falsey that isn't `None`.